test/metabase/test_runner/junit.clj

```diff
@@ -0,0 +1,106 @@
+(ns metabase.test-runner.junit
+  (:require [clojure.test :as t]
+            [metabase.test-runner.junit.write :as write]))
+
+(defmulti ^:private handle-event!*
+  {:arglists '([event])}
+  :type)
+
+(defn handle-event! [{test-var :var, :as event}]
+  (let [test-var (or test-var
+                     (when (seq t/*testing-vars*)
+                       (last t/*testing-vars*)))
+        event    (merge
+                  {:var test-var}
+                  event
+                  (when test-var
+                    {:ns (:ns (meta test-var))}))]
```

Comment:
(for my own edification), is this equivalent to `select-keys`?

Reply:
Close but not exactly equivalent. `(select-keys (meta test-var) [:ns])` would only select the key if `:ns` is present:

```clj
(select-keys {:some-other-key 1} [:ns])
=> {}

{:ns (:ns {:some-other-key 1})}
=> {:ns nil}
```

In this case I don't think one form is better than the other -- either works here.

Every `test-var` should have `:ns` metadata -- the only way a var doesn't end up with `:ns` metadata I think is if you created one by hand (i.e., you're doing weird stuff) whereas all the test vars we're running tests for are created by `deftest` and have `:ns` metadata. So I think the form I used works for now